src/sonic-host-services/scripts/hostcfgd

```diff
@@ -530,6 +571,211 @@ class AaaCfg(object):
+            f.write(pam_passwh_conf)
+        os.chmod(PAM_PASSWORD_CONF + ".tmp", 0o644)
+        os.rename(PAM_PASSWORD_CONF + ".tmp", PAM_PASSWORD_CONF)
+
+        # Age policy
+        # When feature disabled or age policy disabled, expiry days policy should be as linux default, other, accoriding CONFIG_DB.
+        curr_expiration = LINUX_DEFAULT_PASS_MAX_DAYS
+        curr_expiration_warning = LINUX_DEFAULT_PASS_WARN_AGE
+
+        if passw_policies:
+            if passw_policies['state'] == 'enabled':
+                if 'expiration' in passw_policies:
+                    if int(self.passw_policies['expiration']) != 0: # value '0' meaning age policy is disabled
+                        #  the logic is to modify the expiration time according the last updated modificatiion
+                        #
+                        curr_expiration = int(passw_policies['expiration'])
+
+                if 'expiration_warning' in passw_policies:
+                    if int(self.passw_policies['expiration_warning']) != 0: # value '0' meaning age policy is disabled
+                        curr_expiration_warning = int(passw_policies['expiration_warning'])
+
+        if passw_policies:
+            if self.is_passwd_aging_expire_update(curr_expiration):
+                # Set aging policy for existing users
+                self.passwd_aging_expire_modify(curr_expiration)
+
+                # Aging policy for new users
+                self.modify_single_file_inplace(ETC_LOGIN_DEF, ["\'/^PASS_MAX_DAYS/c\PASS_MAX_DAYS " +str(curr_expiration)+"\'"])
+
+            if self.is_passwd_aging_warning_update(curr_expiration_warning):
+                # Aging policy for existing users
+                self.passwd_aging_warn_modify(curr_expiration_warning)
+
+                # Aging policy for new users
+                self.modify_single_file_inplace(ETC_LOGIN_DEF, ["\'/^PASS_WARN_AGE/c\PASS_WARN_AGE " +str(curr_expiration_warning)+"\'"])
+
+    def passwd_aging_expire_modify(self, curr_expiration):
+        normal_accounts = self.get_normal_accounts()
+        if not normal_accounts:
+            syslog.syslog(syslog.LOG_ERR,"failed, no normal users found in /etc/passwd")
+            return
+
+        for normal_account in normal_accounts:
+
+            new_curr_expiration = self.get_user_expire_time(normal_account,curr_expiration)
+
+            try:
+                chage_p_m = subprocess.Popen(('chage', '-M ' + str(new_curr_expiration), normal_account), stdout=subprocess.PIPE)
+                return_code_chage_p_m = chage_p_m.poll()
+                if return_code_chage_p_m != 0:
+                    syslog.syslog(syslog.LOG_ERR, "failed: return code - {}".format(return_code_chage_p_m))
+
+            except subprocess.CalledProcessError as e:
+                syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(e.cmd, e.returncode, e.output))
+
+    def passwd_aging_warn_modify(self, curr_expiration_warning):
```

Comment:
The logic seems to be same in passwd_aging_expire_modify and passwd_aging_warn_modify. Can we use a single function with parameters?